compose/config/types.py

```diff
@@ -14,6 +15,8 @@
 from compose.const import IS_WINDOWS_PLATFORM
 from compose.utils import splitdrive
 
+win32_root_path_pattern = re.compile(r'[A-Za-z]\:\\.*')
```

Comment:
This should perhaps start with `^`, although it arguably doesn't matter (a path with `C:\` somewhere in the middle is clearly invalid anyway).